File: ismgl/lib/data/models/recu_model.dart
class RecuModel {
  final int idRecu;
  final String numeroRecu;
  final int idPaiement;
  final int idEtudiant;
  final double montantTotal;
  final String dateEmission;
  final int emisPar;
  final String? fichierPdf;
  final bool estImprime;
  final String? dateImpression;
  final int nombreImpressions;
  final String? dateCreation;

  // Champs joints (vues API)
  final String? numeroPaiement;
  final String? datePaiement;
  final String? numeroEtudiant;
  final String? nomCompletEtudiant;
  final String? email;
  final String? nomFrais;
  final String? modePaiement;
  final String? referenceTransaction;
  final String? numeroInscription;
  final String? nomFiliere;
  final String? nomNiveau;
  final String? codeAnnee;
  final String? emisParNom;
  final String? pdfUrl;

  RecuModel({
    required this.idRecu,
    required this.numeroRecu,
    required this.idPaiement,
    required this.idEtudiant,
    required this.montantTotal,
    required this.dateEmission,
    required this.emisPar,
    this.fichierPdf,
    this.estImprime = false,
    this.dateImpression,
    this.nombreImpressions = 0,
    this.dateCreation,
    this.numeroPaiement,
    this.datePaiement,
    this.numeroEtudiant,
    this.nomCompletEtudiant,
    this.email,
    this.nomFrais,
    this.modePaiement,
    this.referenceTransaction,
    this.numeroInscription,
    this.nomFiliere,
    this.nomNiveau,
    this.codeAnnee,
    this.emisParNom,
    this.pdfUrl,
  });

  factory RecuModel.fromJson(Map<String, dynamic> json) {
    int asInt(dynamic v) => v is int ? v : int.tryParse(v?.toString() ?? '0') ?? 0;
    String asString(dynamic v) => v?.toString() ?? '';
    return RecuModel(
      idRecu:               asInt(json['id_recu']),
      numeroRecu:           asString(json['numero_recu']),
      idPaiement:           asInt(json['id_paiement']),
      idEtudiant:           asInt(json['id_etudiant']),
      montantTotal:         double.tryParse(json['montant_total']?.toString() ?? '0') ?? 0,
      dateEmission:         asString(json['date_emission']),
      emisPar:              asInt(json['emis_par']),
      fichierPdf:           json['fichier_pdf']          as String?,
      estImprime:           json['est_imprime'] == true || json['est_imprime'] == 1,
      dateImpression:       json['date_impression']      as String?,
      nombreImpressions:    asInt(json['nombre_impressions']),
      dateCreation:         json['date_creation']        as String?,
      numeroPaiement:       json['numero_paiement']      as String?,
      datePaiement:         json['date_paiement']        as String?,
      numeroEtudiant:       json['numero_etudiant']      as String?,
      nomCompletEtudiant:   json['nom_complet_etudiant'] as String?,
      email:                json['email']                as String?,
      nomFrais:             json['nom_frais']            as String?,
      modePaiement:         json['mode_paiement']        as String?,
      referenceTransaction: json['reference_transaction'] as String?,
      numeroInscription:    json['numero_inscription']   as String?,
      nomFiliere:           json['nom_filiere']          as String?,
      nomNiveau:            json['nom_niveau']           as String?,
      codeAnnee:            json['code_annee']           as String?,
      emisParNom:           json['emis_par_nom']         as String?,
      pdfUrl:               json['pdf_url']              as String?,
    );
  }

  Map<String, dynamic> toJson() => {
        'id_recu':          idRecu,
        'numero_recu':      numeroRecu,
        'id_paiement':      idPaiement,
        'id_etudiant':      idEtudiant,
        'montant_total':    montantTotal,
        'date_emission':    dateEmission,
        'emis_par':         emisPar,
        'est_imprime':      estImprime,
        'nombre_impressions': nombreImpressions,
      };
}

File: ismgl-api/models/RecuModel.php
<?php
require_once __DIR__ . '/../core/Model.php';

class RecuModel extends Model {
    protected $table = 'recus';
    protected $primaryKey = 'id_recu';
    
    public function getRecuComplet($id) {
        return $this->db->fetchOne(
            "SELECT r.*, p.numero_paiement, p.date_paiement, p.montant, p.reference_transaction,
                    e.numero_etudiant,
                    CONCAT(u.nom, ' ', u.prenom) as nom_complet_etudiant,
                    u.email, u.telephone,
                    tf.nom_frais,
                    mp.nom_mode as mode_paiement,
                    i.numero_inscription,
                    f.nom_filiere,
                    n.nom_niveau,
                    aa.code_annee,
                    CONCAT(ue.nom, ' ', ue.prenom) as emis_par_nom
             FROM recus r
             JOIN paiements p ON r.id_paiement = p.id_paiement
             JOIN etudiants e ON r.id_etudiant = e.id_etudiant
             JOIN users u ON e.id_user = u.id_user
             JOIN types_frais tf ON p.id_type_frais = tf.id_type_frais
             JOIN modes_paiement mp ON p.id_mode_paiement = mp.id_mode_paiement
             JOIN inscriptions i ON p.id_inscription = i.id_inscription
             JOIN filieres f ON i.id_filiere = f.id_filiere
             JOIN niveaux n ON i.id_niveau = n.id_niveau
             JOIN annees_academiques aa ON i.id_annee_academique = aa.id_annee_academique
             JOIN users ue ON r.emis_par = ue.id_user
             WHERE r.id_recu = ?",
            [$id]
        );
    }
    
    public function getRecuByNumero($numero) {
        return $this->getRecuComplet(
            $this->db->fetchColumn("SELECT id_recu FROM recus WHERE numero_recu = ?", [$numero])
        );
    }
    
    public function getRecuByPaiement($idPaiement) {
        return $this->db->fetchOne(
            "SELECT * FROM recus WHERE id_paiement = ?",
            [$idPaiement]
        );
    }
    
    public function marquerImprime($id) {
        return $this->db->update(
            "UPDATE recus SET est_imprime = TRUE, date_impression = NOW(), nombre_impressions = nombre_impressions + 1 WHERE id_recu = ?",
            [$id]
        );
    }
    
    public function getRecusByEtudiant($idEtudiant) {
        return $this->db->fetchAll(
            "SELECT r.*, p.montant, p.date_paiement, tf.nom_frais
             FROM recus r
             JOIN paiements p ON r.id_paiement = p.id_paiement
             JOIN types_frais tf ON p.id_type_frais = tf.id_type_frais
             WHERE r.id_etudiant = ?
             ORDER BY r.date_emission DESC",
            [$idEtudiant]
        );
    }
}